### PR TITLE
Only update failsafe in release, stage failsafe earlier, fixed SBP USB serial adapter fixes

### PR DESCRIPTION
**Pull request by silverjam**
+ We need to fix the start-up order for the failsafe update script, which needs to access the fpga, so it must be run after fpga init.

+ Stage the failsafe earlier, so that'll actually get included in a release image

+ Fix the gadget mode usb serial adapter that's "SBP only", the naming changes when we're in a release (locked down) image

+ Add a few more helper commands for viewing logs on the device without needing ssh access

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.5.0-develop-2018061520-7-gd6674190
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=554580)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gd6674190&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gd6674190&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.0-develop-2018061520-7-gd6674190
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.5.0-develop-2018061520-7-gd6674190)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.0-develop-2018061520-7-gd6674190/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.0-develop-2018061520-7-gd6674190

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.5.0-develop-2018061520-7-gd6674190
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=554584)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gd6674190&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gd6674190&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.0-develop-2018061520-7-gd6674190
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.5.0-develop-2018061520-7-gd6674190)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.0-develop-2018061520-7-gd6674190/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.0-develop-2018061520-7-gd6674190

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.5.0-develop-2018061520-6-g0f6f7b0b
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=555010)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-6-g0f6f7b0b&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-6-g0f6f7b0b&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.0-develop-2018061520-6-g0f6f7b0b
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.5.0-develop-2018061520-6-g0f6f7b0b)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.0-develop-2018061520-6-g0f6f7b0b/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.0-develop-2018061520-6-g0f6f7b0b

File: board/piksiv3/post_image.sh [hunk 1]
```diff
@@ -89,11 +106,9 @@ generate_dev() {
...
 }
 
-generate_prod() {
-
-  echo -n "Generating PROD firmware image image... "
-
-  cp -v $FAILSAFE_BIN_PATH $FIRMWARE_DIR/PiksiMulti-FAILSAFE.bin
+generate_prod()
+{
+  echo -n ">>> Generating PROD firmware image image... "
```
Comment (benjaminaltieri) [inline board/piksiv3/post_image.sh]: image image

File: package/ports_daemon/ports_daemon/src/ports.c [hunk 2]
```diff
@@ -64,6 +77,20 @@ typedef int (*opts_get_fn_t)(char *buf, size_t buf_size,
 
 #define RUNIT_SERVICE_DIR "/var/run/ports_daemon/sv"
 
+static int opts_get_fixed_sbp(char *buf, size_t buf_size,
+                              const opts_data_t *opts_data)
+{
+  (void) opts_data;
+
+  struct stat s;
+  const char* tty = stat(RELEASE_LOCKDOWN, &s) == 0 ? "ttyGS1" : "ttyGS2";
```
Comment (benjaminaltieri) [inline package/ports_daemon/ports_daemon/src/ports.c]: it's a bummer that we have to do these sorts of gymnastics but good catch on the lockdown enumeration issue and making it work!

File: Makefile [hunk 3]
```diff
@@ -38,22 +39,36 @@ dev-tools-build: pkg-piksi_dev_tools
 
 rel-lockdown-clean: pkg-release_lockdown-dirclean
 
-image-release-open: export BR2_BUILD_RELEASE_OPEN=y
-image-release-open: config
+POST_IMAGE_ENV = HW_CONFIG=prod \
+								 BINARIES_DIR=buildroot/output/images \
```
[the remark this answers is not in the document]
Comment (silverjam) [reply Makefile]: Eh, yeah, editor is using tabs here, I'll try to make it look less silly

[the remark this answers is not in the document]
Comment (silverjam) [reply Makefile]: Makefiles require tabs for rule definitions, but not for vars, so easy fix

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.5.0-develop-2018061520-7-g79957bdc
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=557499)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-g79957bdc&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-g79957bdc&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.0-develop-2018061520-7-g79957bdc
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.5.0-develop-2018061520-7-g79957bdc)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.0-develop-2018061520-7-g79957bdc/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.0-develop-2018061520-7-g79957bdc

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.5.0-develop-2018061520-7-g79957bdc
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=557503)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-g79957bdc&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-g79957bdc&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.0-develop-2018061520-7-g79957bdc
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.5.0-develop-2018061520-7-g79957bdc)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.0-develop-2018061520-7-g79957bdc/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.0-develop-2018061520-7-g79957bdc

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.5.0-develop-2018061520-7-gdfb7f7cb
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=557637)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gdfb7f7cb&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gdfb7f7cb&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.0-develop-2018061520-7-gdfb7f7cb
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.5.0-develop-2018061520-7-gdfb7f7cb)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.0-develop-2018061520-7-gdfb7f7cb/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.0-develop-2018061520-7-gdfb7f7cb

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.5.0-develop-2018061520-7-gdfb7f7cb
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=557641)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gdfb7f7cb&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.5.0-develop-2018061520-7-gdfb7f7cb&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.5.0-develop-2018061520-7-gdfb7f7cb
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.5.0-develop-2018061520-7-gdfb7f7cb)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.5.0-develop-2018061520-7-gdfb7f7cb/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.5.0-develop-2018061520-7-gdfb7f7cb

File: package/ports_daemon/ports_daemon/src/serial.c [hunk 4]
```diff
@@ -21,11 +23,17 @@
...
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <limits.h>
 
 #include <libpiksi/logging.h>
 
 #include "serial.h"
 
+#define RELEASE_LOCKDOWN "/etc/release_lockdown"
```
Comment (benjaminaltieri) [inline package/ports_daemon/ports_daemon/src/serial.c]: do we still need this with the symlink change?

**Pull request merged**